packages/flutter_tools/lib/src/base/features.dart

```diff
@@ -0,0 +1,245 @@
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// The [Feature] for macOS desktop.
+const Feature flutterMacOSDesktopFeature = Feature(
+  name: 'Flutter Desktop for macOS',
+  configSetting: 'flutter-macos-desktop',
+  environmentOverride: 'ENABLE_FLUTTER_DESKTOP',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// The [Feature] for Linux desktop.
+const Feature flutterLinuxDesktopFeature = Feature(
+  name: 'Flutter Desktop for Linux',
+  configSetting: 'flutter-linux-desktop',
+  environmentOverride: 'ENABLE_FLUTTER_DESKTOP',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// The [Feature] for Windows desktop.
+const Feature flutterWindowsDesktopFeature = Feature(
+  name: 'Flutter Desktop for Windows',
+  configSetting: 'flutter-windows-desktop',
+  environmentOverride: 'ENABLE_FLUTTER_DESKTOP',
+  master: FeatureSetting(
+    available: true,
+    enabledByDefault: false,
+  ),
+);
+
+/// A [FeatureFlags] that looks up values based on [Feature] definitions.
+class FlutterFeatureFlags implements FeatureFlags {
+  const FlutterFeatureFlags();
+
+  @override
+  bool get isLinuxEnabled => _isEnabled(flutterLinuxDesktopFeature);
+
+  @override
+  bool get isMacOSEnabled => _isEnabled(flutterMacOSDesktopFeature);
+
+  @override
+  bool get isWebEnabled => _isEnabled(flutterWebFeature);
+
+  @override
+  bool get isWindowsEnabled => _isEnabled(flutterWindowsDesktopFeature);
+
+  // Calculate whether a particular feature is enabled for the current channel.
+  static bool _isEnabled(Feature feature) {
+    final String currentChannel = FlutterVersion.instance.channel;
+    FeatureSetting featureSetting;
+    switch (currentChannel) {
+      case 'stable':
+        featureSetting = feature.stable;
+        break;
+      case 'beta':
+        featureSetting = feature.beta;
+        break;
+      case 'dev':
+        featureSetting = feature.dev;
+        break;
+      case 'master':
+      default:
+        featureSetting = feature.master;
+        break;
+    }
+    if (!featureSetting.available) {
+      return false;
+    }
+    bool isEnabled = featureSetting.enabledByDefault;
+    if (feature.configSetting != null) {
+      final bool configOverride = Config.instance.getValue(feature.configSetting);
+      if (configOverride != null) {
+        isEnabled = configOverride;
+      }
+    }
+    if (feature.environmentOverride != null) {
+      if (platform.environment[feature.environmentOverride] != null) {
+        isEnabled = true;
+      }
+    }
+    return isEnabled;
+  }
+}
+
+/// A [Feature] is process for conditionally enabling tool features.
+///
+/// All settings are optional, and if not provided will generally default to
+/// a "safe" value, such as being off.
+///
+/// The top level feature settings can be provided to apply to all channels.
+/// Otherwise, more specific settings take precidence over higher level
```

Comment:
*precedence*

Reply:
Fixed by removing paragraph (which is no longer relevant).